databricks/koalas/__init__.py

```diff
@@ -20,10 +20,30 @@
 from databricks.koalas.version import __version__  # noqa: F401
 
 
+def assert_python_version():
+    import logging
+
+    _major = 3
+    _minor = 5
+    _deprecated_version = (_major, _minor)
+    _min_supported_version = (_major, _minor + 1)
+
+    if sys.version_info[:2] == _deprecated_version:
+        logging.warning(
+            "Koalas support for Python {dep_ver} is deprecated and will be dropped in "
+            "the future release. At that point, existing Python {dep_ver} workflows "
+            "that use Koalas will continue to work without modification, but Python {dep_ver} "
+            "users will no longer get access to the latest Koalas features and bugfixes. "
+            "We recommend that you upgrade to Python {min_ver} or newer.".format(
+                dep_ver=".".join(map(str, _deprecated_version)),
+                min_ver=".".join(map(str, _min_supported_version)),
+            )
```

Comment:
@ueshin shall we add `FutureWarning` warning type?

Reply:
Currently it's using `logging.warning` as same as the following warnings. Should we use `warnings` instead here?

Reply:
If we're worried it becomes too noisy, we could use `DeprecatedWarning` type too which is not printed out by default.

Reply:
Yeah, if they are deprecation, we should use such types instead of UserWarning